src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/Testing/MsTestV2/MsTestIntegration.cs

```diff
@@ -0,0 +1,158 @@
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Datadog.Trace.Ci;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.ExtensionMethods;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Testing.MsTestV2
+{
+    internal static class MsTestIntegration
+    {
+        internal const string IntegrationName = nameof(IntegrationIds.MsTestV2);
+        internal static readonly IntegrationInfo IntegrationId = IntegrationRegistry.GetIntegrationInfo(IntegrationName);
+        internal static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(MsTestIntegration));
+
+        internal static bool IsEnabled => Common.TestTracer.Settings.IsIntegrationEnabled(IntegrationId);
+
+        internal static Scope OnMethodBegin<TTestMethod>(TTestMethod testMethodInfo, Type type)
+            where TTestMethod : ITestMethod
+        {
+            MethodInfo testMethod = testMethodInfo.MethodInfo;
+            object[] testMethodArguments = testMethodInfo.Arguments;
+
+            string testFramework = "MSTestV2 " + type.Assembly.GetName().Version;
+            string testSuite = testMethodInfo.TestClassName;
+            string testName = testMethodInfo.TestMethodName;
+
+            Scope scope = Common.TestTracer.StartActive("mstest.test", serviceName: Common.TestTracer.DefaultServiceName);
+            Span span = scope.Span;
+
+            span.Type = SpanTypes.Test;
+            span.SetTraceSamplingPriority(SamplingPriority.AutoKeep);
+            span.ResourceName = $"{testSuite}.{testName}";
+            span.SetTag(TestTags.Suite, testSuite);
```

Comment:
This seems like an appropriate time to create a new `TestTags` type that implements `ITags`, but that doesn't need to happen in this PR

Reply:
Yeah all the `ITags` work came after the test frameworks integrations, I'll update the implementation in another PR.